test/functions/ferguson/hijacking/Application8.good

```diff
@@ -1 +1,4 @@
-In Application.Widget.foo
+Application8.chpl:17: In function 'main':
+Application8.chpl:20: error: ambiguous call 'Widget.run(int(64))'
+Application8.chpl:12: note: candidates are: Widget.run(x: real)
```

Comment:
This one feels OK to me since neither candidate is a perfect match.

If I were forced to pick one, I'd pick `Base.run` because `instance` _is_ a `Base` (albeit a more specialized version of `Base`) while `int` is an exact match.  In contrast, for `Application8`, even though `instance` is an exact match, `int` is only coercible to `real` (i.e., it must change / be copied / turn into something else to call that `run` method).  So I feel as though less needs to change in order to call `Base.run()`.

But I think the ambiguity forces the user to be more careful about things which is good (and if we're proven wrong, we can change it into one of the other behaviors later).

Reply:
> I do think subclass methods should take some priority over parent class methods

While I am interested in investigating bigger changes to disambiguation rules than this PR goes for, I really don't want to do it in this PR. But in any case, our current disambiguation rules do prefer subclass methods due to the "if there is an implicit conversion / subtype relationship from T_X to T_X then M_X is more specific" rule. It's just that this preference only applies to the `this` receiver argument and it does not eclipse any preferences for other arguments.